### PR TITLE
Export Diagnostics plus Code Refactor

### PR DESCRIPTION
**Pull request by ljades**
Export Diagnostics Template, plus Laterality addition to code partial template, plus tests, plus some code refactor revolving around these changes.

The refactoring involves augmenting functionality used in past export functionality to improve modularity across the growing custom options for each set of templates.

### CODE REVIEW

**Review by mattrbianchi — APPROVED** [review]
I'm still not the best at reviewing the actual templates, but I like all the refactorings a lot.

File: models/value_set.go [hunk 1]
```diff
@@ -47,12 +47,8 @@ type CodeSet struct {
 	Values []Concept
 }
 
-func (v *ValueSetMap) ReasonValueSetOid(codedValue CodedConcept, fieldOids map[string][]string) string {
-	return v.OidForCode(codedValue, fieldOids["REASON"])
-}
-
-func (v *ValueSetMap) OrdinalityValueSetOid(codedValue CodedConcept, fieldOids map[string][]string) string {
-	return v.OidForCode(codedValue, fieldOids["ORDINAL"])
+func (v *ValueSetMap) OidForCodeForField(codedValue CodedConcept, fieldOids map[string][]string, field string) string {
```
Comment (mattrbianchi) [inline models/value_set.go]: Now that I look at this code, is there a way that we can just change all the calls to this method to use the one below it, OidForCode()? If the caller simply puts the field parameter into the map, the wrapper is unnecessary and OidForCodeForField() can go away.

File: exporter/cat1_view_helper.go [hunk 2]
```diff
@@ -253,20 +262,14 @@ func hasPreferredCode(pc models.Concept) bool {
 	return pc.Code != "" && pc.CodeSystem != ""
 }
 
+func hasLaterality(l models.Laterality) bool {
```
Comment (okeefm) [inline exporter/cat1_view_helper.go:265]: Is there a reasoning behind having `hasLaterality` but getting rid of `hasSetIncisionTime` and `hasSetOrdinality`?

Comment (mattrbianchi) [reply exporter/cat1_view_helper.go:265]: It isn't gone, he moved it to the proper struct it belongs on. Procedure, I think?

**Review by okeefm — APPROVED** [review]

File: exporter/cat1_view_helper.go (hunk 2 above)
Comment (ljades) [reply exporter/cat1_view_helper.go:265]: hasLaterality is more parallel to hasPreferredCode, which are helper functions used for the _code.xml helper function, working on a CodeDisplay object. It's not parallel to the hasSetOrdinality or hasSetSeverity methods, which were ported to the relevant HasEntry interface implementations.

**Pull request merged**